crates/noirc_evaluator/src/ssa_refactor/ir/instruction/call.rs

```diff
@@ -0,0 +1,334 @@
+        Intrinsic::SlicePushFront => {
+            let slice = dfg.get_array_constant(arguments[0]);
+            if let (Some((mut slice, element_type)), elem) = (slice, arguments[1]) {
+                slice.push_front(elem);
+                let new_slice = dfg.make_array(slice, element_type);
+                SimplifyResult::SimplifiedTo(new_slice)
+            } else {
+                SimplifyResult::None
+            }
+        }
+        Intrinsic::SlicePopBack => {
+            let slice = dfg.get_array_constant(arguments[0]);
+            if let Some((mut slice, element_type)) = slice {
+                let elem =
+                    slice.pop_back().expect("There are no elements in this slice to be removed");
+                let new_slice = dfg.make_array(slice, element_type);
+                SimplifyResult::SimplifiedToMultiple(vec![new_slice, elem])
+            } else {
+                SimplifyResult::None
+            }
+        }
+        Intrinsic::SlicePopFront => {
+            let slice = dfg.get_array_constant(arguments[0]);
+            if let Some((mut slice, element_type)) = slice {
+                let elem =
+                    slice.pop_front().expect("There are no elements in this slice to be removed");
+                let new_slice = dfg.make_array(slice, element_type);
+                SimplifyResult::SimplifiedToMultiple(vec![elem, new_slice])
+            } else {
+                SimplifyResult::None
+            }
+        }
+        Intrinsic::SliceInsert => {
+            let slice = dfg.get_array_constant(arguments[0]);
+            let index = dfg.get_numeric_constant(arguments[1]);
+            if let (Some((mut slice, element_type)), Some(index), value) =
+                (slice, index, arguments[2])
+            {
+                slice.insert(index.to_u128() as usize, value);
+                let new_slice = dfg.make_array(slice, element_type);
+                SimplifyResult::SimplifiedTo(new_slice)
+            } else {
+                SimplifyResult::None
+            }
+        }
+        Intrinsic::SliceRemove => {
+            let slice = dfg.get_array_constant(arguments[0]);
+            let index = dfg.get_numeric_constant(arguments[1]);
+            if let (Some((mut slice, element_type)), Some(index)) = (slice, index) {
+                let removed_elem = slice.remove(index.to_u128() as usize);
+                let new_slice = dfg.make_array(slice, element_type);
+                SimplifyResult::SimplifiedToMultiple(vec![new_slice, removed_elem])
+            } else {
+                SimplifyResult::None
+            }
+        }
+        Intrinsic::BlackBox(bb_func) => simplify_bb_func(bb_func, arguments, dfg),
+        Intrinsic::Println | Intrinsic::Sort => SimplifyResult::None,
+    }
+}
+
+/// Try to simplify this black box call. If the call can be simplified to a known value,
+/// that value is returned. Otherwise [`SimplifyResult::None`] is returned.
+fn simplify_bb_func(
```

Comment:
```suggestion
fn simplify_black_box_func(
```
I always read `bb` as `barretenberg`